src/tools/miri/src/shims/intrinsics/mod.rs

```diff
@@ -131,6 +131,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 this.write_pointer(Pointer::new(ptr.provenance, masked_addr), dest)?;
             }
 
+            // The default implementation always returns `false`, but we want
+            // to return either `true` or `false` at random.
+            "is_comptile_time_known" => {
```

Comment:
Please also add a test calling this intrinsic to `src/tools/miri/tests/pass/intrinsics.rs`.